core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -1234,74 +1241,30 @@ public int compare(FingerprintFacet o1, FingerprintFacet o2) {
     }
 
     /**
-     * Save the settings to a file.
+     * Save the Fingerprint in the Fingerprint Storage
      * @throws IOException Save error
      */
     public synchronized void save() throws IOException {
-        if(BulkChange.contains(this))   return;
+        if(BulkChange.contains(this)) return;
 
         long start=0;
         if(logger.isLoggable(Level.FINE))
             start = System.currentTimeMillis();
 
-        File file = getFingerprintFile(md5sum);
-        save(file);
-        SaveableListener.fireOnChange(this, getConfigFile(file));
+        FingerprintStorage.get().save(this);
```

Comment:
I would say that it is a Jenkins API issue. `SaveableListener` is equipped to work only with the Filesystem storage. It might be out of the scope for this project, so I raised https://issues.jenkins-ci.org/browse/JENKINS-62543 . It would be great to add a TODO comment to the code so that visitors

```suggestion
        // Implementations are expected to invoke SaveableListener on their own if relevant
        // TODO: Consider improving Saveable Listener API: https://issues.jenkins-ci.org/browse/JENKINS-62543
        FingerprintStorage.get().save(this);
```